mmv1/templates/terraform/constants/datastream_stream.go.erb

```diff
@@ -0,0 +1,66 @@
+<%- # the license inside this block applies to this file
+  # Copyright 2022 Google Inc.
+  # Licensed under the Apache License, Version 2.0 (the "License");
+  # you may not use this file except in compliance with the License.
+  # You may obtain a copy of the License at
+  #
+  #     http://www.apache.org/licenses/LICENSE-2.0
+  #
+  # Unless required by applicable law or agreed to in writing, software
+  # distributed under the License is distributed on an "AS IS" BASIS,
+  # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  # See the License for the specific language governing permissions and
+  # limitations under the License.
+-%>
+func resourceDatastreamStreamCustomDiffFunc(diff TerraformResourceDiff) error {
+	if diff.HasChange("desired_state") {
+		old, new := diff.GetChange("desired_state")
+		oldState := old.(string)
+		newState := new.(string)
+
+		if isNewResource(diff) {
+			if newState != "NOT_STARTED" && newState != "RUNNING" {
+				return fmt.Errorf("`desired_state` can only be set to `NOT_STARTED` or `RUNNING` when creating a new Stream")
+			}
+		} else {
+			if newState == "NOT_STARTED" && oldState != newState {
+				return fmt.Errorf("Field `desired_state` can only be set to `NOT_STARTED` when creating a new Stream")
```

Comment:
from datastream perspective a new stream can be created in RUNNING state

Reply:
I've clarified the error message - this is only relevant to already-created streams.

We could also handle this by forcing the resource to be recreated in this case - but I don't know if that is a desirable behavior or not, & we could make that change later.

Reply:
ok, sg